src/CMakeLists.txt

```diff
@@ -39,6 +39,10 @@ find_package(BZip2  QUIET)
 # ----------------------------------------------------------------------------
 # App-Level Configuration
 # ----------------------------------------------------------------------------
+if (DEFINED ENV{CI})
+    option (DR_YARA_NATIVE_BUILD "Architecture-specific optimizations (disable on CI)" OFF)
+endif ()
```

Comment:
Also could have passed `-DR_YARA_NATIVE_BUILD=OFF` to the cmake in the CI. Now this behaviour is kinda hidden in the CMake instead of being obvious in the workflow.

Now local builds will use `native` by default, while CI builds will not; this might cause trouble in the future